infrastructure/merklemountainrange/src/merkle_change_tracker.rs

```diff
@@ -230,3 +228,101 @@ impl MerkleChangeTracker {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::mmr::*;
+    use blake2::Blake2b;
+    use digest::Digest;
+    use serde_derive::{Deserialize, Serialize};
+    use std::fs;
+    use tari_storage::lmdb::*;
+    use tari_utilities::Hashable;
+
+    #[derive(Serialize, Deserialize)]
+    pub struct IWrapper(u32);
+
+    impl Hashable for IWrapper {
+        fn hash(&self) -> Vec<u8> {
+            Blake2b::new().chain(self.0.to_le_bytes()).result().to_vec()
+        }
+    }
+
+    fn create_mmr(leaves: u32) -> MerkleMountainRange<IWrapper, Blake2b> {
+        let mut mmr: MerkleMountainRange<IWrapper, Blake2b> = MerkleMountainRange::new();
+        mmr.init_persistance_store(&"mmr".to_string(), 5);
+        for i in 1..leaves + 1 {
+            let object: IWrapper = IWrapper(i);
+            mmr.push(object);
+        }
+        mmr
+    }
+
+    #[test]
+    fn create_med_mmr() {
+        let _res = fs::remove_dir_all("./tests/test_mmr_cm"); // we ensure that the test dir is empty
+        let mut mmr = create_mmr(14);
+        // create storage
+        fs::create_dir("./tests/test_mmr_cm").unwrap();
+        let builder = LMDBBuilder::new();
+        let mut store = builder
+            .set_mapsize(5)
+            .set_path("./tests/test_mmr_cm/")
+            .add_database(&"mmr_mmr_checkpoints".to_string())
+            .add_database(&"mmr_mmr_objects".to_string())
+            .add_database(&"mmr_init".to_string())
+            .build()
+            .unwrap();
+        let result = mmr.apply_checkpoint(&mut store);
+        assert_eq!(result.is_ok(), true);
```

Comment:
Can replace ln 278 and 279 with ```assert!(mmr.apply_checkpoint(&mut store).is_ok());```, there are a few places where this can be done.